pkg/registry/memory/nse_server.go

```diff
@@ -45,44 +61,37 @@ func (n *networkServiceEndpointRegistryServer) Find(query *registry.NetworkServi
 		return err
 	}
 	if query.Watch {
-		go func() {
-			eventCh := make(chan *registry.NetworkServiceEndpoint, n.eventChannelSize)
-			n.eventChannelsLocker.Lock()
-			n.eventChannels = append(n.eventChannels, eventCh)
-			n.eventChannelsLocker.Unlock()
-			err := sendAllMatches(query.NetworkServiceEndpoint)
-			if err != nil {
-				return
-			}
-			for {
-				select {
-				case event := <-eventCh:
-					if matchutils.MatchNetworkServiceEndpoints(query.NetworkServiceEndpoint, event) {
-						if err := s.Send(event); err != nil {
-							return
-						}
+		eventCh := make(chan *registry.NetworkServiceEndpoint, n.eventChannelSize)
+		n.eventChannelsLocker.Lock()
+		n.eventChannels = append(n.eventChannels, eventCh)
+		n.eventChannelsLocker.Unlock()
+		err := sendAllMatches(query.NetworkServiceEndpoint)
```

Comment:
Should we combine next.NetworkServiceEndpointRegistryServer(s.Context()).Find(query, s) in case of watch somehow?